.github/workflows/pr_master_test.yml

```diff
@@ -442,11 +448,46 @@ jobs:
             echo "Uploading coverage to codecov (attempt ${i})"
             bash $CODECOV -Z -X gcov -X s3 -f coverage.xml
             if test $? == 0; then
+                echo "PASS" > ${GITHUB_WORKSPACE}/codecov.result
+                break
+            elif test $i -ge 2; then
+                # Do not fail the build just because the codecov upload fails
+                echo "FAIL" > ${GITHUB_WORKSPACE}/codecov.result
                 break
-            elif test $i -ge 5; then
-                exit 1
             fi
             DELAY=$(( RANDOM % 30 + 30))
             echo "Pausing $DELAY seconds before re-attempting upload"
             sleep $DELAY
         done
+
+    - name: Record build artifacts
```

Comment:
I'm 50/50 on whether we should archive. My hesitancy centers around storage limits - I'm fairly certain that artifacts contribute to an organization's total storage limit. From what I've read, they get cleaned up automatically every 90 days, and there is a community-created job that can trigger cleanup sooner: https://github.com/c-hive/gha-remove-artifacts <- So just something to keep in mind.

Reply:
FWIW, it looks like we're archiving around 55MB for each suite completed. Our limit is 500MB.

Reply:
Good call.  I put it in because some time I wanted to write code to analyze all our runs and see if there are tests that are being skipped everywhere.  But there is no reason to *always* archive the XML: if we ever get around to writing that analysis, we can just turn it back on for a [WIP] PR, collect the results, and never merge.